src/content/en/updates/posts/2016/06/webrtc-ecdsa.markdown

```diff
@@ -0,0 +1,69 @@
+}
+@media screen and (max-width: 500px) {
+  img.screenshot {
+    max-width: 100%;
+  }
+}
+</style>
+
+<p class="intro">You can now choose which type of cryptographic certificate to use for setting up a WebRTC peer connection.</p>
+
+Until recently, WebRTC used RSA-1024 keys for certificates. There are several disadvantages with these keys:
+
+* Generating RSA-1024 keys adds around 1000ms to call setup time.
+* 1024-bit RSA keys do not provide adequate cryptographic strength.
+
+Because certificate generation with RSA-1024 is slow, some sites have resorted to preparing certificates in advance or reusing them.
+
+The key strength issue could be resolved by going to 2048-bit RSA keys or more, but that would delay call setup by several additional seconds. Instead of changing the RSA key size, Chrome 52 implements ECDSA keys (Elliptic Curve Digital Signature Algorithm) for use in certificates. These are as strong as 3072-bit RSA keys‚ but several thousand times faster: call setup overhead with ECDSA is just a few milliseconds.
+
+> Breaking an RSA key requires you to factor a large number. We are pretty good at factoring large numbers and [getting better all the time](http://bristolcrypto.blogspot.co.uk/2013/02/discrete-logarithms.html). Breaking an ECDSA key requires you to solve the Elliptic Curve Discrete Logarithm Problem (ECDLP). The mathematical community has not made any major progress in improving algorithms to solve this problem since is was independently introduced by Koblitz and Miller in 1985.
+
+— [Nick Sullivan](https://blog.cloudflare.com/ecdsa-the-digital-signature-algorithm-of-a-better-internet/), CloudFlare
+
+All in all, ECDSA keys mean better security, better privacy and better performance — especially on mobile. For these reasons, ECDSA has been mandated in the [WebRTC Security Architecture](https://www.ietf.org/mail-archive/web/rtcweb/current/msg14754.html) draft.
+
+From Chrome 47 you can opt in to ECDSA:
+
+{% highlight javascript %}
+pc.generateCertificate({name: "ECDSA", namedCurve: "P-256"})
+{% endhighlight %}
+
+From Chrome 52, ECDSA is enabled by default.
+
+## Storing RTCCertificate in IndexedDB
+
+Another improvement in Chrome 52: the RTCCertificates used by WebRTC can be saved and loaded from IndexedDB storage, avoiding the need to generate new certificates between sessions.
+
+RTCCertificates are self-signed certificates used in the [DTLS](https://en.wikipedia.org/wiki/Datagram_Transport_Layer_Security) handshake when setting up a WebRTC peer connection. (DTLS is an implementation of the cryptographic protocol [TLS](https://en.wikipedia.org/wiki/Transport_Layer_Security) for datagram protocols such as [UDP](https://en.wikipedia.org/wiki/User_Datagram_Protocol), which is used by WebRTC.) RTCCertificates can be stored by implementing [structured cloning](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm), a new algorithm for serializing JavaScript objects.
+
```

Comment:
The first half of this paragraph seems to belong in the intro (what certificates are).
The second half seems like an implementation detail seen from the users' perspective. Drop it?

The main use I can see for IndexedDB at the moment is avoiding the RSA generation overhead if you still have the need to use RSA; another for advanced users is certificate continuity, but I'm not certain it's possible to write an app that takes advantage of that yet.